utils/crypto.cpp

```diff
@@ -178,6 +183,111 @@ namespace
             return false;
         }
     }
+
+
+    size_t
+    write_function(void* contents,size_t size, size_t nmemb, void* userp)
+    {
+        reinterpret_cast<std::string*>(userp)->append(reinterpret_cast<char*>(contents), size * nmemb);
+        return size * nmemb;
+    }
+
+
+    bzn::json_message
+    get_curl_response(const std::string& function_call, const std::string& url = ROPSTEN_URL)
+    {
+        const std::string MSG_ERROR_CURL{"curl_easy_perform() failed: "};
+        std::unique_ptr<CURL, std::function<void(CURL*)>> curl(curl_easy_init(),
+                                                               std::bind(curl_easy_cleanup, std::placeholders::_1));
+        std::string readBuffer;
+
+        curl_easy_setopt(curl.get(), CURLOPT_WRITEDATA, &readBuffer);
+
+        const std::string post_fields{boost::str(boost::format(REQUEST_BASE) % function_call)};
+
+        curl_easy_setopt(curl.get(), CURLOPT_POSTFIELDS, post_fields.c_str());
+
+        curl_easy_setopt(curl.get(), CURLOPT_POSTFIELDSIZE, -1L);
+        curl_easy_setopt(curl.get(), CURLOPT_URL, url.c_str());
+        curl_easy_setopt(curl.get(), CURLOPT_WRITEFUNCTION, reinterpret_cast<void*>(write_function));
+
+        CURLcode res = curl_easy_perform(curl.get());
+        if (res != CURLE_OK)
+        {
+            throw (std::runtime_error(MSG_ERROR_CURL + curl_easy_strerror(res)));
```

Comment:
Is this truly a fatal error?

Reply:
Probably not. I think I can just do what I do in the net block. I'll move response up a bit...